lib/api/v3/work_packages/work_packages_by_project_api.rb

```diff
@@ -0,0 +1,68 @@
+#-- copyright
+# OpenProject is a project management system.
+# Copyright (C) 2012-2015 the OpenProject Foundation (OPF)
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See doc/COPYRIGHT.rdoc for more details.
+#++
+
+require 'api/v3/work_packages/work_package_representer'
+require 'api/v3/work_packages/work_packages_shared_helpers'
+
+module API
+  module V3
+    module WorkPackages
+      class WorkPackagesByProjectAPI < ::API::OpenProjectAPI
+        resources :work_packages do
+          helpers ::API::V3::WorkPackages::WorkPackagesSharedHelpers
+
+          post do
+            service = create_service
+            work_package = service.create
```

Comment:
I'd rather write the `create_service` as `@foo ||= bar` instead of only `bar`.

Thereby you would cache the result and could rewrite the lines using the service to

````ruby
create_service.create
# some other stuff
create_service.save # this is still the same service that created your WP
````